Twenty Twelve: Move skip link to be first focusable element

himanshupathak95 opened the pull request:
Trac ticket: https://core.trac.wordpress.org/ticket/62967

This PR fixes an accessibility issue in the Twenty Twelve theme by moving the "Skip to content" link to be the first focusable element on the page. No changes to the link's functionality or styling.

himanshupathak95 committed:
Twenty Twelve: Move skip link to be first focusable element

himanshupathak95 committed:
Twenty Twelve: Fix skip link visibility on focus

sabernhardt commented on src/wp-content/themes/twentytwelve/header.php:
```diff
@@ -33,6 +33,7 @@
 
 <body <?php body_class(); ?>>
 <?php wp_body_open(); ?>
+<a class="assistive-text" href="#content"><?php _e( 'Skip to content', 'twentytwelve' ); ?></a>
 <div id="page" class="hfeed site">
 	<header id="masthead" class="site-header">
```
```suggestion
<div id="page" class="hfeed site">
	<a class="screen-reader-text skip-link" href="#content"><?php _e( 'Skip to content', 'twentytwelve' ); ?></a>
	<header id="masthead" class="site-header">
```

1. I prefer placing the link inside the "page" `div` (as Twenty Fifteen and some later bundled themes do). If someone uses a link to `#page`, then the skip link would be the first item.
2. If you also change the `assistive-text` class to `screen-reader-text`, Twenty Twelve already has styles for `.site .screen-reader-text:focus`. Then `style.css` would not need any editing.

himanshupathak95 committed:
Twenty Twelve: Move the skip link inside the page div

himanshupathak95 committed:
Twenty Twelve: Revert styles changes

himanshupathak95 commented:
Hey @sabernhardt Thanks for the suggestions!

I have also updated the changes and moved the skip links inside the page `div` for the [Twenty Thirteen](https://github.com/WordPress/wordpress-develop/pull/8327) and [Twenty Fourteen](https://github.com/WordPress/wordpress-develop/pull/8326) themes. Please let me know if the changes look good to you whenever you have time. 

Thanks.

himanshupathak95 commented:
Merged in the changeset - [[59912]](https://core.trac.wordpress.org/changeset/59912/)

sabernhardt commented:
I planned to gather screenshots, and they look fine in Twenty Twelve.

Before r59912:
![skip links in top corner](https://github.com/user-attachments/assets/037ae7f5-a960-4125-84ca-6c81b7e7a361)

Twenty Twelve now inherits a few styles from `block-library/style.css`, and the skip links are still in the top corner.
```
line-height: normal;
text-decoration: none;
margin: -1px;
```
![skip links in same position but slightly different styling](https://github.com/user-attachments/assets/62b3f7c8-c690-419f-a168-c9638bbc277e)
